unmsmfisi-socialapplication/social_app
# [DEVOPS] Implement frontend terraform pipeline

this aims for implementing the CD deployment of the iac

**Purpose:**
The purpose of this workflow is to automate the deployment of infrastructure for a frontend application using Terraform. It ensures consistency, reliability, and efficiency in managing infrastructure changes.

**How to Test:**
To test this workflow, you can follow these steps:

- Ensure Secrets: Make sure that the necessary secrets (GCP_TEST_TERRAFORM_KEY, TEST_PROJECT_ID, FRONTEND_TF_VARS, TEST_BUCKET_TF_NAME) are set in your GitHub repository.
- Manual Trigger: The workflow is set to run manually (workflow_dispatch). Trigger the workflow manually in the GitHub Actions tab to observe its execution.
- Review Logs: Examine the logs produced by each step in the GitHub Actions workflow to ensure that each step completes successfully.
- Validation: Check if the Terraform validation step passes, ensuring that the Terraform configurations are valid.
